### DISCUSSION TITLE
Running cross compiled Bun App failing on startup due to migration issue

### DISCUSSION THREAD

**Discussion by dylanhillier** [uptrace/bun#820] [Q&A]
I have a bun application i have been developing on my Mac with the intent to deploy it to an Ubuntu server. As such, i have built it using:

`env GOOS=linux GOARCH=amd64 go build -o myapp cmd/bun/main.go`

After copying it over to the ubuntu server the app will not run because it is unable to find migrations. I have tried copying over the migrations and putting them in various different directories but have had no luck. 

On execution i see an error like this:

```
ubuntu@ip-0-0-0-0:~$ ./myapp -env=dev api
panic: stat .: no such file or directory

goroutine 1 [running]:
backend/cmd/bun/migrations.init.0()
        /Users/me/Development/personal/myapp/backend/cmd/bun/migrations/main.go:9 +0x3f
```

Is this (cross compiling a bun app) just something that will never work, or is there a way to work around this issue?

I do realist i could just build it on an Ubuntu server, or on mac in a container, but i'd rather avoid that if i can.

**Comment by vmihailenco** [comment]
You can try to embed migrations in the binary: https://github.com/uptrace/uptrace/blob/master/pkg/bunapp/pgmigrations/main.go#L11-L12

Or figure out how current working dir works on Linux.

**Reply by dylanhillier** [reply]
Thanks @vmihailenco . I will give it a try over the next few days.
